src/uwtools/tests/test_rocoto.py

```diff
@@ -44,26 +45,65 @@ def test_add_jobname():
     assert expected == tree
 
 
-def test_write_rocoto_xml(tmp_path):
-    input_yaml = support.fixture_path("hello_workflow.yaml")
-    with resources.as_file(resources.files("uwtools.resources")) as resc:
-        input_template = resc / "rocoto.jinja2"
+def test__add_tasks():
+    with resources.as_file(resources.files("uwtools.tests.fixtures")) as path:
+        input_yaml = (path / "hello_workflow.yaml").as_posix()
+
+    values = YAMLConfig(input_yaml)
+    tasks = values["workflow"]["tasks"]
+    with patch.object(rocoto, "_add_jobname") as module:
+        rocoto._add_tasks(input_yaml)
+    assert module.called_once_with(tasks)
+
+
+def test__rocoto_template():
+    with resources.as_file(resources.files("uwtools.resources")) as path:
+        expected = (path / "schema_with_metatasks.rng").as_posix()
+    assert rocoto._rocoto_template() == expected
+
+
+def test__rocoto_schema():
+    with resources.as_file(resources.files("uwtools.resources")) as path:
+        expected = (path / "rocoto.jsonschema").as_posix()
+    assert rocoto._rocoto_schema() == expected
+
+
+@pytest.mark.parametrize("vals", [("hello_workflow.yaml", True), ("fruit_config.yaml", False)])
+def test_realize_rocoto_xml(vals, tmp_path):
+    fn, validity = vals
     output = tmp_path / "rendered.xml"
-    rocoto.write_rocoto_xml(
-        input_yaml=input_yaml, input_template=str(input_template), rendered_output=str(output)
-    )
 
-    expected = support.fixture_path("hello_workflow.xml")
-    support.compare_files(expected, output)
+    with patch.object(rocoto, "validate_rocoto_xml", value=True):
+        with patch.object(rocoto.uwtools.config.validator, "_bad_paths", return_value=None):
+            with resources.as_file(resources.files("uwtools.tests.fixtures")) as path:
+                input_yaml = (path / fn).as_posix()
+                result = rocoto.realize_rocoto_xml(input_yaml=input_yaml, rendered_output=output)
+    assert result is validity
+
+
+def test_realize_rocoto_invalid_xml():
```

Comment:
It looks like the last outstanding issue is why this test is not successfully covering `rocoto.py` [lines 98-99](https://github.com/WeirAE/workflow-tools/blob/aa912672d1b16dabc4ca3e8d0fc439a1c31912db/src/uwtools/rocoto.py#L98)

Reply:
Fixed it - had to fix my `with patch.object()` usage